exercises/exercise_3/README.md

```diff
@@ -1,266 +1,255 @@
-
-* **Deploy** using `docker stack deploy`, the `-` references the standard input
-```sh
-$ docker-app render hello | docker stack deploy my-hello-app -c -
-Creating network my-hello-app_default
-Creating service my-hello-app_hello
-```
-* **Open** a browser at the `8080` port, it displays `Hello DockerCon`
-* **Re-deploy** changing the text
-```sh
-$ docker-app render hello -s hello.text="Hello Moby" | docker stack deploy my-hello-app -c -
-```
-* **Refresh** your browser and check the message
-* **Deploy** another application using the production parameters, it should open it on port `80`
-```sh  
-$ docker-app render hello -f prod-parameters.yml | docker stack deploy my-prod-app -c -
-```
-* **Remove** the two stacks using `docker stack rm`
-
-**NOTE:** With the parameters, you can now easily scale your application with `docker stack`.
-
-**Bonus Exercise:** Add parameterized `deploy.replicas` to the `hello` service and use it to scale up and down with the `render` command and `--set hello.replicas=X`
-
-## Summary
-
-- With parameters you can use the same compose file and target multiple environments (dev/test/staging/prod/...) 
-- Render let's you check what will be the exact compose-file deployed
-- You can pipe the result of a render to your current workflow (`docker-compose`, `docker stack deploy`) and still use some of the benefits of docker-app
-
-Here is a generic application package template:
-
-```yaml
-version: 0.1.0
-name: base
-description: A generic application template which takes an image as a parameter
-maintainers:
-  - name: garethr
-    email: garethr@docker.com
-
----
-version: '3.7'
-services:
-  app:
-    image: ${image}
-    ports:
-     - ${port}:${port}
-    deploy:
-      replicas: ${replicas}
-      resources:
-        limits:
-          memory: ${memory}
-
----
-image: 
-port: 8080
-replicas: 1
-memory: 128MB
-```
+# Exercise - Running and Sharing our Docker Application
+
+> **Time**: Approximately 10 minutes
+>
+> **Difficulty**: Easy
+
+## Table of Contents
+
+1. [Deploying the Docker App](#deploying-the-docker-app)
+1. [Viewing Installed Applications](#viewing-installed-applications)
+1. [Uninstalling the Docker App](#uninstalling-the-docker-app)
+
+
+## Exercise Objectives
+
+By the end of this exercise, you will have:
+
+- Deployed the Docker App using the locally-sourced app bundle
+- Learned how to see existing app installations
+- Removed an app installation
+- Tagged and pushed the Docker App to Docker Hub
+
+
+## Deploying the Docker App
+
+There are two different ways Docker App can locate an application bundle
+
+- **Locally** - Docker App will use a merged or split application bundle definition found on your local filesystem
+- **Remote** - Docker App will pull the application bundle from Docker Hub (or another registry)
+
+For our first deployment, we will use source from the locally available app definitions.
```

Comment:
> we will use source from

Isn't it missing something? `we will use the sources` ? Or maybe I'm misreading it?